### PR TITLE
Switch to the latest Tuned daemon.

### PR DESCRIPTION
**Pull request by jmencak**
Changes
  - Updated to the latest upstream Tuned daemon
    https://github.com/redhat-performance/tuned
    23dcbd7ead35048735a6e2d291f385b48330b5ab
  - Dropped all 2.14 patches as they are all included
    upstream
  - Added an e2e test for the new feature of the [scheduler]
    plugin default_irq_smp_affinity
  - Removed unnecessary quoting in test/e2e/util/util.go

### CODE REVIEW

File: test/e2e/basic/default_irq_smp_affinity.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	ntoconfig "github.com/openshift/cluster-node-tuning-operator/pkg/config"
+	util "github.com/openshift/cluster-node-tuning-operator/test/e2e/util"
+)
+
+// Test the application (and rollback) of [scheduler]'s default_irq_smp_affinity option.
+// See: https://github.com/redhat-performance/tuned/pull/306
+var _ = ginkgo.Describe("[basic][default_irq_smp_affinity] Node Tuning Operator set irq default smp affinity", func() {
+	const (
+		profileAffinity           = "../testing_manifests/default_irq_smp_affinity.yaml"
+		nodeLabelAffinity         = "tuned.openshift.io/default-irq-smp-affinity"
```
Comment (marcel-apf) [inline test/e2e/basic/default_irq_smp_affinity.go:22]: This label is only used bu the test, right?

File: test/e2e/testing_manifests/default_irq_smp_affinity.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  profile:
+  - data: |
+      [main]
+      summary=An OpenShift profile to test [scheduler] default_irq_smp_affinity option
+      include=openshift-node
+      [scheduler]
+      # isolated_cores take a list of ranges; e.g. isolated_cores=2,4-7
+      isolated_cores=1
+      # the cpulist in 'default_irq_smp_affinity' is unpacked and written directly to /proc/irq/default_smp_affinity
+      default_irq_smp_affinity=1
```
Comment (marcel-apf) [inline test/e2e/testing_manifests/default_irq_smp_affinity.yaml:16]: OK, that sets the default_irq_smp_affinity differently from the default functionality of the tuned.
I would add a test ensuring the prev functionality still works the same (not setting default_irq_smp_affinity results in the inverse mask of the isolated_cores)

File: Makefile [hunk 3]
```diff
@@ -14,7 +14,7 @@ REV=$(shell git describe --long --tags --match='v*' --always --dirty)
 
 # Upstream tuned daemon variables
 TUNED_REPO:=https://github.com/redhat-performance/tuned.git
-TUNED_TAG:=v2.14.0
+TUNED_COMMIT:=23dcbd7ead35048735a6e2d291f385b48330b5ab
```
[the remark this answers is not in the document]
Comment (jmencak) [reply Makefile:17]: AFAIU there will be a release candidate `v2.15.0-rc.1` in cca 2 weeks.  That is probably too late.  We can try asking on the tuned IRC channel, but targetting a commit should be a bit more flexible (read independent of upstream Tuned daemon).

File: test/e2e/basic/default_irq_smp_affinity.go (hunk 1 above)
Comment (jmencak) [reply test/e2e/basic/default_irq_smp_affinity.go:22]: Yes. it needs to match `default_irq_smp_affinity.yaml`.  I should probably add a comment.

File: test/e2e/testing_manifests/default_irq_smp_affinity.yaml (hunk 2 above)
Comment (jmencak) [reply test/e2e/testing_manifests/default_irq_smp_affinity.yaml:16]: I'll see what I can do.

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcel-apf) [reply Makefile:17]: agreed if 1.15.X is on the way there is nothing to ask for

File: test/e2e/basic/default_irq_smp_affinity.go (hunk 1 above)
Comment (marcel-apf) [reply test/e2e/basic/default_irq_smp_affinity.go:22]: thanks

File: test/e2e/testing_manifests/default_irq_smp_affinity.yaml (hunk 2 above)
Comment (marcel-apf) [reply test/e2e/testing_manifests/default_irq_smp_affinity.yaml:16]: is only a thought, nothing more

**Comment by jmencak** [comment]
/hold
The affinity calculation for the e2e tests needs more work.

File: test/e2e/testing_manifests/default_irq_smp_affinity.yaml (hunk 2 above)
Comment (jmencak) [reply test/e2e/testing_manifests/default_irq_smp_affinity.yaml]: Done.

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (jmencak) [reply Makefile:17]: See the discussion on the tuned IRC channel.  There should be a release candidate next week.  Depends whether you want to wait.

**Review by marcel-apf — APPROVED** [review]
/lgtm

Thanks,
Marcel

**Comment by jmencak** [comment]
Thank you for the review, @marcel-apf !

**Pull request merged**